### ISSUE TITLE
exports: interface does not work for inline interfaces

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#3959]
### Version Information

* vyper Version (output of `vyper --version`): e1adb7b3344c1ac03facfa553830a94dd7def2e2

### What's your issue about?

`exports: interface` does not work for interfaces which were declared inline. example:
```vyper
# lib.vy
from ethereum.ercs import IERC20

asset: public(immutable(IERC20))
 
interface IAsset:
    def asset() -> address: view

implements: IAsset

@deploy
def __init__(asset_: IERC20):
    asset = asset_
```

```vyper
# main.vy

from ethereum.ercs import IERC20

import lib
initializes: lib

exports: lib.IAsset

@deploy
def __init__(asset_: IERC20):
    lib.__init__(asset_)
```

**Comment by charles-cooper** [comment]
mildly related: https://github.com/vyperlang/vyper/issues/3954

**Comment by charles-cooper** [comment]
fixed in #4090
